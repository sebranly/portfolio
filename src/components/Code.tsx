import * as React from 'react';

export interface CodeProps {
  className?: string;
  lines: string[];
}

const Code: React.FC<CodeProps> = (props) => {
  const { className, lines } = props;

  return (
    <code className={className}>
      {lines.map((line: string, index: number) => {
        return <pre key={index}>{line}</pre>;
      })}
    </code>
  );
};

export { Code };
