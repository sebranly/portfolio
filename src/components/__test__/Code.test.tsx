import * as React from 'react';
import { render } from '@testing-library/react';

import { Code } from '../Code';

test('Code', () => {
  const lines = [
    '+---+---+---+---+',
    '| . | Q | . | . |',
    '+---+---+---+---+',
    '| . | . | . | Q |',
    '+---+---+---+---+',
    '| Q | . | . | . |',
    '+---+---+---+---+',
    '| . | . | Q | . |',
    '+---+---+---+---+'
  ];
  
  const createComponent = () => <Code lines={lines} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
