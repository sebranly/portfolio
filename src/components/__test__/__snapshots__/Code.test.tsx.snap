// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code 1`] = `
NodeList [
  <code>
    <pre>
      +---+---+---+---+
    </pre>
    <pre>
      | . | Q | . | . |
    </pre>
    <pre>
      +---+---+---+---+
    </pre>
    <pre>
      | . | . | . | Q |
    </pre>
    <pre>
      +---+---+---+---+
    </pre>
    <pre>
      | Q | . | . | . |
    </pre>
    <pre>
      +---+---+---+---+
    </pre>
    <pre>
      | . | . | Q | . |
    </pre>
    <pre>
      +---+---+---+---+
    </pre>
  </code>,
]
`;
